ReducersManager implementation and rework of the LocalStorage

gf-isi opened the pull request.

phev8 commented:
- Does this rework require any change how the library is imported or configured in the existing apps? 
- Could you please provide a short description for the changelog?

gf-isi commented:
No, it should not require any change in the client application. Nothing public should have changed.
Beware though, that in some web application I've seen something like 

`import { RootState } from 'case-web-app-core/build/store/rootReducer';`

This is possibly a bad practice because `RootState` is not publicly available since it is not explicitly exported from an index. The changes in this request should still be compatible, but some tests might be required if other non public member are imported in this way. 

Here is a brief list of changes

- Added the possibility to add reducers to the store through the `ReducersManager`
- Added the possibility to manage how the state is saved to the `localStorage` through the `LocalStoreManager`
- Added an event being dispatched at the end of the registration workflow

Everything should be documented in the usage section of the doc.
